conbench/templates/app.html

```diff
@@ -157,26 +157,46 @@ <h4 class="modal-title"><span class="glyphicon glyphicon-trash submenu"></span>&
             var that = this;
             var reset = ""
             var mySelectList = $("<select class='custom-search-box'></select>")
+                //stops ordering event trigger when clicked on a menu selection
+                .on("click", function (e){
+                    e.stopImmediatePropagation()
+                })
                 .appendTo(this.header())
                 .on("change", function() {
-                    that.search($(this).val()).draw();
+                    currentValue = $(this).val()
+                    //search for empty string value for the selected column
+                    if(currentValue === ""){
+                        that.search('^$', true, false).draw();
+                    }
+                    //reset search for the selected column
+                    else if(currentValue == "All"){
+                        that.search('').draw();
+                    }
+                    // select for non-empty values
+                    else{
+                        that.search(currentValue).draw();
+                    }
+                    console.log(currentValue)
                 });
             mySelectList.append(
-                $(`<option value="${reset}">${reset}</option>`)
+                $(`<option value="All">${reset}</option>`)
             );
             this.cache("search").unique().sort().each(function(param) {
-                // check for multiple whitespaces and empty string
-                if (/^\s*$/.test(param) == false) {
+                //trim the whitespaces (setting to empty string) before appending to drop-down list
+                param = param.trim()
```

Comment:
this trims leading and trailing whitespace, yes?

Reply:
Yup :)